### PR TITLE
DSL2: Convert input BAM

### LINKED ISSUE #987: DSL2: Add input BAM BAM2FASTQ conversion support for remapping
As discussed
```[tasklist]
### Tasks
- [ ] https://github.com/nf-core/eager/issues/999
```

### PR DESCRIPTION
**Pull request by TCLamnidis**
Closes #987 

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
    - [ ] If you've added a new tool - add to the software_versions process and a regex to `scrape_software_versions.py`
    - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](<https://github.com/>nf-core/eager/tree/master/.github/CONTRIBUTING.md)
    - [ ] If necessary, also make a PR on the nf-core/eager _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint .`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: workflows/eager.nf [hunk 1]
```diff
@@ -94,6 +96,53 @@ workflow EAGER {
...
+            .filter {
+                meta, reads ->
+                meta.single_end
+            }
+            .join(SAMTOOLS_CONVERT_INPUT_BAM.out.fastq_other)
+            .map {
+                meta, read1, fastq_other ->
+                [meta, [read1, fastq_other] ]
+            }
+            .dump(tag: "converted_se_and_other", pretty: true)
```
Comment (jfy133) [inline workflows/eager.nf:118]: Remove debugging dumps everywhere
```suggestion
```

**Review by jfy133 — APPROVED** [review]
I think it makes sense to me... I follow the logic at least, I guess it'll be alldown to testing!

**Comment by TCLamnidis** [comment]
Made some changes to skip using the input BAMs if they get converted, and tweaked spaces where output from the now-optional modules was used.

Have another look just for completion.

**Comment by TCLamnidis** [comment]
Waiting for James to test specifically that the right reads from the BAM make it to the FastQs. Not updated the branch to make testing easier. 

James, once you are done and happy, feel free to update and merge

**Review by jfy133 — COMMENTED** [review]
Testing works for me :+1:

File: conf/modules.config [hunk 2]
```diff
@@ -34,6 +34,25 @@ process {
...
 
+    //
+    // CONVERT INPUT BAM
+    //
+    withName: SAMTOOLS_CONVERT_INPUT_BAM {
+        tag = { "${meta.sample_id}_${meta.library_id}_L${meta.lane}" }
+        ext.prefix = { "${meta.sample_id}_${meta.library_id}_L${meta.lane}" }
+        publishDir = [
+            enabled: false
+        ]
```
[the remark this answers is not in the document]
Comment (TCLamnidis) [reply conf/modules.config:45]: This is how it is in `2.*`, with the converted reads never being published. Could add an option to keep them, though I don't see why anyone would 😅

[the remark this answers is not in the document]
Comment (TCLamnidis) [reply conf/modules.config:45]: I will merge now and if someone requests these be published, we can add that later.

**Pull request merged**